components/client-attestation-mgt/org.wso2.carbon.identity.client.attestation.mgt/src/main/java/org/wso2/carbon/identity/client/attestation/mgt/validators/AndroidAttestationValidator.java

```diff
@@ -0,0 +1,288 @@
+            // The generated Integrity token is considered old, likely due to a replay attack.
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Attestation request provided by Client :" + applicationResourceId +
+                        " in tenant : " + tenantDomain + " is older than required window.");
+            }
+            clientAttestationContext.setAttested(false);
+            clientAttestationContext.setValidationFailureMessage("Package name in the request details does " +
+                    "not match with the requested client.");
+            return false;
+        } else {
+            // Request details are valid.
+            return true;
+        }
+    }
+
+    /**
+     * Validates the integrity of the application based on the recognition verdict provided in
+     * a DecodeIntegrityTokenResponse.
+     *
+     * @param decodeIntegrityTokenResponse The response containing the integrity token and
+     *                                     application integrity details.
+     * @param clientAttestationContext  The context to store the validation results and updated information.
+     * @return True if the application is recognized as "PLAY_RECOGNIZED"; false otherwise.
+     * recognition verdict is not as expected.
+     */
+    private boolean validateAppIntegrity(DecodeIntegrityTokenResponse decodeIntegrityTokenResponse,
+                                         ClientAttestationContext clientAttestationContext) {
+
+        // Extract application integrity details from the response.
+        AppIntegrity appIntegrity = decodeIntegrityTokenResponse.getTokenPayloadExternal().getAppIntegrity();
+
+        // Check if the application is recognized as "PLAY_RECOGNIZED."
+        if (StringUtils.equals(appIntegrity.getAppRecognitionVerdict(), PLAY_RECOGNIZED)) {
+            return true;
+        }
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Attestation request provided by Client :" + applicationResourceId + " in tenant : "
+                    + tenantDomain + " is invalid. Application integrity validation failed."
+                    + " Unexpected recognition verdict: " + appIntegrity.getAppRecognitionVerdict());
+        }
+        clientAttestationContext.setAttested(false);
+        clientAttestationContext.setValidationFailureMessage("Application integrity validation failed." +
+                " Unexpected recognition verdict: " + appIntegrity.getAppRecognitionVerdict());
+        // Throw an exception with a descriptive message indicating that the application
+        // integrity verdict is unexpected.
+        return false;
+    }
+
+    private long getConfiguredAllowedWindow() {
+
+        String allowedWindow = IdentityUtil.getProperty(CLIENT_ATTESTATION_ALLOWED_WINDOW_IN_MILL_SECOND);
+        if (StringUtils.isNotEmpty(allowedWindow)) {
+            return Long.parseLong(allowedWindow);
+        } else {
+            return 60000L; // 1 minute
+        }
```

Comment:
Since we have a default validity period configured in the server lets not impose a code-level default. If the server value is empty let's skip the validation.